### ISSUE TITLE
Issue with foldFreeAp

### ISSUE THREAD

**Issue by LiamGoodacre** [ethul/purescript-freeap#3]
`analyzeFreeAp` appears to append monoid values the wrong way round to what I expect.

Here's a rough example:

``` purescript
data M r = A r | B r

printM :: forall a. FreeAp M a -> String
printM = analyzeFreeAp go where
  go (A v) = "A"
  go (B v) = "B"

ma = liftFreeAp (A unit)
mb = liftFreeAp (B unit)
main = log (printM (ma *> mb *> mb))
```

The output here is `BBA` when I was expecting `ABB`

**Comment by LiamGoodacre** [comment]
Actually I think the issue is with `foldFreeAp`/`apply` - investigating...

**Comment by ethul** [comment]
Closed via #4
